src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,145 @@
+                () -> {
+                registrationService.register(NULL_AGE_USER); },
+                "It should throw RegistrationException when age is null");
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_PASSWORD_USER); },
+                "Password length must be more than "
+                        + SHORT_PASSWORD_USER.getPassword().length() + "letters");
+        String passwordOneLetter = "a";
+        SHORT_PASSWORD_USER.setPassword(passwordOneLetter);
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_PASSWORD_USER); },
+                "Password length must be more than " + passwordOneLetter.length() + "letters");
+        String passwordThreeLetters = "abc";
+        SHORT_PASSWORD_USER.setPassword(passwordThreeLetters);
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_PASSWORD_USER); },
+                "Password length must be more than " + passwordThreeLetters.length() + "letters");
+        String passwordFiveLetters = "abcde";
+        SHORT_PASSWORD_USER.setPassword(passwordFiveLetters);
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_PASSWORD_USER); },
+                "Password length must be more than " + passwordFiveLetters.length() + "letters");
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_LOGIN_USER); },
+                "Login length must be more than "
+                        + SHORT_LOGIN_USER.getLogin().length() + "letters");
+        String loginOneLetter = "a";
+        SHORT_LOGIN_USER.setLogin(loginOneLetter);
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_LOGIN_USER); },
+                "Login length must be more than " + loginOneLetter.length() + "letters");
+        String loginThreeLetters = "abc";
+        SHORT_LOGIN_USER.setLogin(loginThreeLetters);
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_LOGIN_USER); },
+                "Login length must be more than " + loginThreeLetters.length() + "letters");
+        String loginFiveLetters = "abcde";
+        SHORT_LOGIN_USER.setLogin(loginFiveLetters);
+        assertThrows(RegistrationException.class,
+                () -> {
+                registrationService.register(SHORT_LOGIN_USER); },
+                "Login length must be more than " + loginFiveLetters.length() + "letters");
+    }
+
+    @Test
+    void register_validUser_Ok() {
```

Comment:
add test for already existing user in storage

and add method that clears storage after each test